transaction/summary/src/verifier.rs

```diff
@@ -117,7 +43,17 @@ pub fn verify_tx_summary(
 /// implementation details of the mc-crypto-digestible scheme.
 /// If TxSummary digestible annotations are changed then this object's
 /// implementation needs to change also.
-pub struct TxSummaryStreamingVerifier {
+pub struct TxSummaryStreamingVerifier<const RECORDS: usize = MAX_RECORDS> {
+    // The report which we produce about what balance changes occur for what
+    // parties
+    report: TxSummaryUnblindingReport<RECORDS>,
+
+    // Context for streaming verifier
+    ctx: TxSummaryStreamingVerifierCtx,
+}
+
+/// Context for computing streaming transaction summaries
+pub struct TxSummaryStreamingVerifierCtx {
```

Comment:
maybe this should also have a `MAX_RECORDS` parameter (and like phantom-data I guess?) to avoid mix and matching report sizes across the different api calls to this function?

not sure, maybe it's nicer the way you did it

Reply:
ahh i also preferred this at the top level too, iirc got in a fight with clippy over trying to hold a const generic without actually _using_ it / binding the global N to the report argument N and couldn't convince `PhantomData` to hold it...

can revisit and see if there's a better option, alternatively we could abstract report into a trait so the verifier doesn't have visibility into the storage at all?

Reply:
no strong preference, happy to dodge a compiler fight 😄